sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -84,14 +84,48 @@ int pollSocketUntilEventOrTimeout(
   return WSAPoll(&poller, 1, timeout);
 #endif
 }
+
+#ifdef WINDOWS
+// Windows needs this after every write to socket or peformance would be reduced to 1/4 for
+// uploading operation.
+// https://github.com/Azure/azure-sdk-for-cpp/issues/644
+void WinSocketSetBuffSize(curl_socket_t socket, std::function<void(std::string const& message)> logger)
```

Comment:
Please follow up with curl (maybe file an issue) to see if this fix can be incorporated into the codebase for things other than just `transfer`.

Reply:
Also, ask what the purpose of `ideallen` is and if it was intended to be used somewhere.